src/Avalonia.Controls/WindowBase.cs

```diff
@@ -42,7 +42,7 @@ public class WindowBase : TopLevel
 
         private bool _hasExecutedInitialLayoutPass;
         private bool _isActive;
-        private bool _ignoreVisibilityChange;
+        protected bool IgnoreVisibilityChange { get; set; }
```

Comment:
Why exactly do you need to make it protected if you override Show and Hide anyway?

Reply:
I think its because of HandleClosed method.

It calls base.HandleClosed, so the flag needs to be shared.

Ill make it a disposable thing.